packages/lib/bookingSuccessRedirect.ts

```diff
@@ -78,8 +80,8 @@ export const useBookingSuccessRedirect = () => {
       navigateInTopWindow(url.toString());
       return;
     }
-    const newSearchParams = getNewSeachParams({ query });
-    return router.push(`/booking/${booking.uid}?${newSearchParams.toString()}`);
+    const newSearchParams = getNewSeachParams({ query, searchParams: searchParams ?? undefined });
+    return router.push(`/booking/${booking.uid}${isEmbed ? "/embed" : ""}?${newSearchParams.toString()}`);
```

Comment:
Passing all query params as is and ensuring that /embed is present. So, that the appropriate header can be added(either by middleware or by withEmbedSSR)